src/js/Controllers/UIController.js

```diff
@@ -786,31 +795,31 @@ class UIController {
     onKeepContext(alert, isSubtle) {
         clearTimeout(this.timers[alert.msgID])
         this.onButtonPress(alert.appID, alert.buttonID, alert.buttonName)
-        var timeout = alert.duration ? alert.duration : 10000
+        const timeout = store.getState().system.resetPeriod;
         const state = store.getState()
         const context = state.activeApp
         
         this.timers[alert.msgID] = setTimeout(this.onAlertTimeout, timeout, alert.msgID, alert.appID, context ? context : alert.appID, isSubtle);
-        this.onResetTimeout(alert.appID, isSubtle ? "UI.SubtleAlert" : "UI.Alert");
+        this.onResetTimeout(alert.msgID, isSubtle ? "UI.SubtleAlert" : "UI.Alert", timeout);
     }
     onSliderKeepContext(msgID, appID, duration) {
         clearTimeout(this.timers[msgID])
         
-        let timeout = duration ? duration : 10000
+        const timeout = store.getState().system.resetPeriod;
         const state = store.getState();
         const context = state.activeApp
 
         this.timers[msgID] = setTimeout(this.onSliderClose, timeout, msgID, appID, context, "TIMEOUT");
-        this.onResetTimeout(appID, "UI.Slider")
+        this.onResetTimeout(msgID, "UI.Slider", timeout);
     }
     onScrollableMessageKeepContext(msgID, appID, duration) {
         clearTimeout(this.timers[msgID]);
-        var timeout = duration ?? 10000;
+        const timeout = store.getState().system.resetPeriod;
```

Comment:
```suggestion
        const timeout = duration ?? 10000;
```

This value is passed in

Reply:
@iCollin please find the changes [here](https://github.com/LuxoftSDL/generic_hmi/pull/39/files).